Camara Brasileira do Livro ISBN.js

```diff
@@ -0,0 +1,760 @@
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectSearch(items) {
+	items = cleanData(items);
+	return !!items.length;
+}
+
+async function doSearch(items) {
+	items = cleanData(items);
+	for (let { ISBN } of items) {
+		let search = ISBN;
+		if (ISBN.length == 10) {
+			search += ' OR ' + ZU.toISBN13(ISBN);
+		}
+		let body = {
+			count: true,
+			facets: [],
+			filter: '',
+			orderby: null,
+			queryType: 'full',
+			search,
+			searchFields: 'FormattedKey,RowKey',
+			searchMode: 'any',
+			select: '*',
+			skip: 0,
+			top: 1
+		};
+		let response = await requestJSON('https://isbn-search-br.search.windows.net/indexes/isbn-index/docs/search?api-version=2016-09-01', {
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/json; charset=UTF-8',
+				'api-key': '100216A23C5AEE390338BBD19EA86D29',
+				Origin: 'https://www.cblservicos.org.br',
+				Referer: 'https://www.cblservicos.org.br/'
+			},
+			body: JSON.stringify(body)
+		});
+		let results = response.value;
+		for (let result of results) {
+			translateResult(result);
+		}
+	}
+}
+
+function translateResult(result) {
+	let item = new Zotero.Item('book');
+	item.title = result.Title;
+	if (result.Subtitle && !item.title.includes(':') && !result.Subtitle.includes(':')) {
+		item.title += ': ' + result.Subtitle;
+	}
+	item.abstractNote = result.Sinopse;
+	// TODO: Need example data for:
+	// item.series
+	// item.seriesNumber
+	// item.volume
+	// item.numberOfVolumes
+	item.edition = result.Edicao;
+	if (item.edition == '1') {
+		item.edition = '';
+	}
+	item.place = result.Cidade;
+	item.publisher = result.Imprint;
+	item.date = ZU.strToISO(result.Date);
+	item.numPages = result.Paginas;
+	if (item.numPages == '0') {
+		item.numPages = '';
+	}
+	item.language = (result.IdiomasObra && result.IdiomasObra[0]) || 'pt-BR';
+	if (item.language == 'português (Brasil)') {
+		item.language = 'pt-BR';
+	}
+	item.ISBN = ZU.cleanISBN(result.FormattedKey);
+	for (let [i, author] of result.Authors.entries()) {
+		if (author == author.toUpperCase()) {
+			author = ZU.capitalizeName(author);
+		}
+		let creatorType;
+		if (result.Profissoes && result.Profissoes.length) {
+			// If there's a corresponding entry in the Profissoes array, use it
+			if (result.Profissoes.length > i) {
+				switch (result.Profissoes[i]) {
+					case 'Coordenador':
```

Comment:
Just to flag that "Coordenador" is a tricky term and I've seen it used as both editor (in the sense of convener) and author. Probably nothing much we can do about that, though.